src/vs/editor/browser/controller/textAreaHandler.ts

```diff
@@ -242,6 +242,17 @@ export class TextAreaHandler extends ViewPart {
 							return new TextAreaState(textBefore, textBefore.length, textBefore.length, position, position);
 						}
 					}
+					// on macOS, write current selection into textarea will allow system text services pick selected text
+					// https://github.com/microsoft/vscode/issues/27799#issuecomment-1200088452
+					if (platform.isMacintosh && !selection.isEmpty()) {
```

Comment:
I am a little bit concerned about performance here, maybe I have a 2 GB text file open and select all. This would mean we end up converting 2 GB of text into a string only then to `substring` it.

Could we do some heuristics and check how large the selection is and not do this if the number of selected text is already exceeding our limit? There maybe API to figure out the selection size efficiently, or we just guess it.

Tbh I would not expect this to work with a substring of my selection, I would rather expect the text area to be empty when we hit the limit, otherwise the user may be very confused why only parts of the text are taken over?

Reply:
+1 make sure to keep in mind the performance limiation of Chromium for large text areas
Issue with more details on Chromium side https://bugs.chromium.org/p/chromium/issues/detail?id=1076525

Reply:
via https://github.com/microsoft/vscode/pull/156717/commits/8086f29bc26154eb2bbae8df8d9f41a4f5451dec